File: lab3/MatrixMul/MatrixComparator.cs
namespace MatrixMul;

static class MatrixComparator
{
    public static double CalcMaxAbsDiff(double[,] A, double[,] B)
    {
        if (A.GetLength(0) != B.GetLength(0) ||
            A.GetLength(1) != B.GetLength(1))
            throw new ArgumentException(
                "Can't compare matrices:\n" +
                $"A {A.GetLength(0)} X {A.GetLength(1)} and " +
                $"B {B.GetLength(0)} X {B.GetLength(1)}"
                );
        double res = 0;
        for (int ir = 0; ir < A.GetLength(0); ir++)
            for (int ic = 0; ic < A.GetLength(1); ic++) {
                var absDiff = Math.Abs(A[ir, ic] - B[ir, ic]);
                if (absDiff > res) {
                    res = absDiff;
                }
            }
        return res;
    }
}

File: lab4/matrix-mul-fork-join/src/main/java/app/Utils/MatrixComparator.java
package app.Utils;

public class MatrixComparator {

    public static double calcMaxAbsDiff(double[][] A, double[][] B) {
        if (A.length != B.length || A[0].length != B[0].length) {
            throw new IllegalArgumentException(
                    "Can't compare matrices:\n" +
                            "A " + A.length + " X " + A[0].length + " and " +
                            "B " + B.length + " X " + B[0].length);
        }

        double res = 0;
        for (int ir = 0; ir < A.length; ir++) {
            for (int ic = 0; ic < A[0].length; ic++) {
                double absDiff = Math.abs(A[ir][ic] - B[ir][ic]);
                if (absDiff > res) {
                    res = absDiff;
                }
            }
        }
        return res;
    }
}
